pika/channel.py

```diff
@@ -753,127 +756,132 @@ def open(self):
-                         {'queue': queue})
+            condition = (spec.Queue.DeclareOk, {'queue': queue})
         else:
-            condition = spec.Queue.DeclareOk  # pylint: disable=R0204
-        replies = [condition] if nowait is False else []
-        self._validate_channel_and_callback(callback)
+            condition = spec.Queue.DeclareOk
+        replies = [condition] if not nowait else []
+
         return self._rpc(spec.Queue.Declare(0, queue, passive, durable,
                                             exclusive, auto_delete, nowait,
                                             arguments or dict()),
                          callback, replies)
 
     def queue_delete(self,
-                     callback=None,
                      queue='',
                      if_unused=False,
                      if_empty=False,
-                     nowait=False):
+                     callback=None):
         """Delete a queue from the broker.
 
-        :param callable callback: The callback to call on Queue.DeleteOk;
-            MUST be None when nowait=True.
         :param queue: The queue to delete
         :type queue: str or unicode
         :param bool if_unused: only delete if it's unused
         :param bool if_empty: only delete if the queue is empty
-        :param bool nowait: Do not wait for a Queue.DeleteOk
+        :param callable callback: The callback to call on Queue.DeleteOk
+        :raises ValueError:
 
         """
-        replies = [spec.Queue.DeleteOk] if nowait is False else []
-        self._validate_channel_and_callback(callback)
+        self._validate_channel()
+        nowait = self._validate_rpc_completion_callback(callback)
+        replies = [spec.Queue.DeleteOk] if not nowait else []
         return self._rpc(spec.Queue.Delete(0, queue, if_unused, if_empty,
                                            nowait),
                          callback, replies)
 
-    def queue_purge(self, callback=None, queue='', nowait=False):
+    def queue_purge(self, queue='', callback=None):
         """Purge all of the messages from the specified queue
 
-        :param callable callback: The callback to call on Queue.PurgeOk;
-            MUST be None when nowait=True.
         :param queue: The queue to purge
         :type queue: str or unicode
-        :param bool nowait: Do not expect a Queue.PurgeOk response
+        :param callable callback: The callback to call on Queue.PurgeOk
+        :raises ValueError:
 
         """
-        replies = [spec.Queue.PurgeOk] if nowait is False else []
-        self._validate_channel_and_callback(callback)
+        self._validate_channel()
+        nowait = self._validate_rpc_completion_callback(callback)
+        replies = [spec.Queue.PurgeOk] if not nowait else []
         return self._rpc(spec.Queue.Purge(0, queue, nowait), callback, replies)
 
     def queue_unbind(self,
-                     callback=None,
                      queue='',
```

Comment:
`queue` shouldn't have a default value since you always have to name the queue to be unbound